### PR TITLE
coreutils: use SRI hash format

### PR DESCRIPTION
**Pull request by AndersonTorres**
Also, some reformatting.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/coreutils/default.nix [hunk 1]
```diff
@@ -10,27 +19,27 @@
...
+in
 stdenv.mkDerivation (rec {
-  pname = "coreutils${optionalString (!minimal) "-full"}";
+  pname = "coreutils${lib.optionalString (!minimal) "-full"}";
   version = "9.1";
 
   src = fetchurl {
     url = "mirror://gnu/coreutils/coreutils-${version}.tar.xz";
-    sha256 = "sha256:08q4b0w7mwfxbqjs712l6wrwl2ijs7k50kssgbryg9wbsw8g98b1";
+    sha256 = "sha256-YaH0ENeLp+fzelpPUObRMgrKMzdUhKMlXt3xejhYBCM=";
```
Comment (mweinelt) [inline pkgs/tools/misc/coreutils/default.nix:38]:
```suggestion
    hash = "sha256-YaH0ENeLp+fzelpPUObRMgrKMzdUhKMlXt3xejhYBCM=";
```

Comment (AndersonTorres) [reply pkgs/tools/misc/coreutils/default.nix:38]: By some strange reason it does not accept the `hash` argument...

**Comment by mohe2015** [comment]
Our of interest: Why does this cause rebuilds?

File: pkgs/tools/misc/coreutils/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/coreutils/default.nix:38]: Has the boot fetchurl hash agility?

File: pkgs/tools/misc/coreutils/default.nix [hunk 2]
```diff
@@ -60,12 +69,12 @@ stdenv.mkDerivation (rec {
...
     # intermittent failures on builders, unknown reason
     sed '2i echo Skipping du basic test && exit 77' -i ./tests/du/basic.sh
-  '' + (optionalString (stdenv.hostPlatform.libc == "musl") (lib.concatStringsSep "\n" [
+  '' + (lib.optionalString (stdenv.hostPlatform.libc == "musl") (lib.concatStringsSep "\n" [
     ''
       echo "int main() { return 77; }" > gnulib-tests/test-parse-datetime.c
       echo "int main() { return 77; }" > gnulib-tests/test-getlogin.c
     ''
-  ])) + (optionalString stdenv.isAarch64 ''
+  ])) + (lib.optionalString stdenv.isAarch64 ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/coreutils/default.nix:77]:
```suggestion
  ]) + lib.optionalString stdenv.isAarch64 ''
```
FYI these are optional.

**Comment by dasJ** [comment]
Honestly, what's the point? Why not wait for the next update where the rebuilds are done anyway?

**Comment by dasJ** [comment]
Also, I don't feel good at all if you haven't even built it (the PR descriptions says so) and it doesn't fit CONTRIBUTING

**Comment by AndersonTorres** [comment]
> Honestly, what's the point? Why not wait for the next update where the rebuilds are done anyway?

It is not a problem to me.
(Indeed I was expecting it would generate a mass rebuild.)

> Also, I don't feel good at all if you haven't even built it (the PR descriptions says so) and it doesn't fit CONTRIBUTING

I will put it on Draft, then.

File: pkgs/tools/misc/coreutils/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/tools/misc/coreutils/default.nix:77]: I merely kept them.

**Comment by SuperSandro2000** [comment]
The mass rebuild is at least caused by the whitespace change in postPatch.

**Comment by AndersonTorres** [comment]
I am cogitating reopening this PR specifically.

**Pull request closed without merging**